client/src/typegenService.ts

```diff
@@ -49,14 +49,13 @@ const throttledTypegenCreationMachine = createMachine<
         await Promise.all([
           Object.entries(context.eventMap).map(async ([, event]) => {
             const uri = event.uri;
-
-            const newUri = vscode.Uri.file(
-              uri.replace(/\.([j,t])sx?$/, ".typegen.ts"),
+            const pathFromUri = vscode.Uri.parse(uri, true).path.slice(1);
```

Comment:
Could you tell us how does your `uri` exactly looks like? this could help us reviewing this change and comparing with unix-style paths

Reply:
You can see the different variables in this image : 

![image](https://user-images.githubusercontent.com/23745106/151660768-fa4b1a49-c154-4cd5-92a2-883ad1f863cf.png)

Reply:
When I remove the `slice(1)` from line 52, it throws an error when trying to write the file. `C:` is added somehow. 

![image](https://user-images.githubusercontent.com/23745106/151660959-9e13c2fa-fcac-4048-9f40-8b2d023a79c2.png)

Reply:
I tried my changes on a debian OS. It didn't work because I was removing the leading slash. I added a check to remove the leading slash only on Windows. I don't know if there is a cleaner way to handle this but now it works for both Windows and Unix